File: csharp/auth/AuthClientCredentials.cs
using System.Net;
using System.Text.Json.Nodes;

namespace csharp.auth;

public class AuthClientCredentials
{
    private readonly HttpClient _client;
    private readonly string _tokenUrl;
    private readonly JwtHandler _jwtHandler;

    public AuthClientCredentials(string tokenUrl, string privateKeyFile, string clientId, string kid, HttpClient? client = null)
    {
        _tokenUrl = tokenUrl;
        _client = client ?? new HttpClient();
        _jwtHandler = new JwtHandler(privateKeyFile, tokenUrl, clientId, kid);
    }

    public async Task<string?> AccessToken(int expInMinutes = 1)
    {
        var jwt = _jwtHandler.GenerateJwt(expInMinutes);
        Console.WriteLine(jwt);
        
        var values = new Dictionary<string, string>
        {
            {"grant_type", "client_credentials"},
            {"client_assertion_type", "urn:ietf:params:oauth:client-assertion-type:jwt-bearer"},
            {"client_assertion", jwt}
        };
        var content = new FormUrlEncodedContent(values);
        
        var response = await _client.PostAsync(_tokenUrl, content);
        var resBody1 = await response.Content.ReadAsStringAsync();
        Console.WriteLine(resBody1);
        if (response.StatusCode != HttpStatusCode.OK)
        {
            throw new Exception("Authentication failed. \n" + response.Content);
        }

        var resBody = await response.Content.ReadAsStringAsync();
        var parsed = JsonNode.Parse(resBody);
        
        return parsed?["access_token"]?.ToString();
    
    }
}

File: application-restricted-signed-jwt-tutorials/php/auth/AuthClientCredentials.php
<?php

namespace php\auth;

require_once "JwtHandler.php";

class AuthClientCredentials
{
  private $tokenUrl;
  private $privateKeyFile;
  private $clientId;
  private $kid;

  function __construct($tokenUrl, $privateKeyFile, $clientId, $kid)
  {
    $this->tokenUrl = $tokenUrl;
    $this->privateKeyFile = $privateKeyFile;
    $this->clientId = $clientId;
    $this->kid = $kid;
  }

  public function AccessToken()
  {
    $jwtHandler = new JwtHandler($this->privateKeyFile, $this->tokenUrl, $this->clientId, $this->kid);
    $jwt = $jwtHandler->GenerateJwt();

    $headers = array(
        "Content-Type: application/x-www-form-urlencoded"
    );

    $data = array(
        "grant_type" => "client_credentials",
        "client_assertion_type" => "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
        "client_assertion" => $jwt
    );

    $curl = curl_init($this->tokenUrl);
    curl_setopt($curl, CURLOPT_URL, $this->tokenUrl);
    curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($curl, CURLOPT_POSTFIELDS, http_build_query($data));
    curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($curl, CURLOPT_HTTPHEADER, $headers);

    $resp = curl_exec($curl);
    curl_close($curl);
    $json = json_decode($resp);
    $accessToken = $json->access_token;
    return $accessToken;
  }
}
